### PR TITLE
SYCL: Alternative solution to avoid runtime error of launching kernel on xpu:1 when que…

### PR DESCRIPTION
**Pull request by fengyuan14**
…rying SYCL kernel bundle ahead
The kernel won't be built for devices except for the first device. Launching kernel on devices except for the first device will raise runtime error. Here is an alternative as a temporary solution to provide an extra hint to SYCL runtime. https://github.com/intel/llvm/issues/15127

### CODE REVIEW

File: src/comm/DeviceProperties.h [hunk 1]
```diff
@@ -16,8 +16,8 @@ static int64_t syclMaxWorkGroupSize(
   auto dev = q.get_device();
 
   auto kid = ::sycl::get_kernel_id<KernelClass>();
-  auto kbundle =
-      ::sycl::get_kernel_bundle<::sycl::bundle_state::executable>(ctx, {kid});
+  auto kbundle = ::sycl::get_kernel_bundle<::sycl::bundle_state::executable>(
+      ctx, {dev}, {kid});
```
Comment (EikanWang) [inline src/comm/DeviceProperties.h:20]: @fengyuan14 , could you help submit a GitHub issue to github.com/intel/llvm and then add some comments here to link the issue to be submitted?

Overall, LGTM.

**Comment by EikanWang** [comment]
@fengyuan14 , by the way, please give some detailed information here.

File: .github/workflows/_linux_ut.yml [hunk 2]
```diff
@@ -94,25 +94,36 @@ jobs:
           python -c "import torch; print(torch.__config__.show())"
           python -c "import torch; print(torch.__config__.parallel_info())"
           python -c "import torch; print(torch.__config__.torch.xpu.device_count())"
-      - name: Run XPU OP Examples
-        if: contains(inputs.ut, 'op_example') || github.event_name == 'schedule'
+      - name: Run XPU OP Regressions test
+        if: contains(inputs.ut, 'op_regression') || github.event_name == 'schedule'
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:98]: Where are those test case come from? From previously example?

File: .github/workflows/_linux_ut.yml [hunk 3]
```diff
@@ -94,25 +94,36 @@ jobs:
...
           pip install pytest
-          timeout 8000 pytest -v 
+          timeout 8000 pytest -v
+      - name: Run XPU OP Regressions test on device 1
+        if: contains(inputs.ut, 'op_regression_dev1') || github.event_name == 'schedule'
+        run: |
+          cd ${{ github.workspace }}
+          xpu-smi discovery
+          source .github/scripts/env.sh
+          # source activate xpu_op_${ZE_AFFINITY_MASK}
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:114]:
```suggestion
          source activate xpu_op_${ZE_AFFINITY_MASK}
          export ZE_AFFINITY_MASK_OLD=${ZE_AFFINITY_MASK}
          unset ZE_AFFINITY_MASK
```

File: .github/workflows/_linux_ut.yml [hunk 4]
```diff
@@ -94,25 +94,36 @@ jobs:
...
+          cd ${{ github.workspace }}
+          cd test/regressions
+          pip install pytest
+          timeout 8000 pytest -v test_operation_on_device_1.py
```
Comment (chuanqi129) [inline .github/workflows/_linux_ut.yml:118]:
```suggestion
          timeout 8000 pytest -v test_operation_on_device_1.py
          export ZE_AFFINITY_MASK=${ZE_AFFINITY_MASK_OLD}
```

File: test/regressions/test_operation_on_device_1.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import torch
+from torch.testing._internal.common_utils import TestCase
+
+class TestOperationOnDevice1(TestCase):
+    def test_sum_on_device1(self, dtype=torch.float):
+        if torch.xpu.device_count() >= 2:
```
Comment (chuanqi129) [inline test/regressions/test_operation_on_device_1.py:6]: For single card node, this test will be skipped by default, right?

**Comment by chuanqi129** [comment]
Please also modify UT scope description in: 
https://github.com/intel/torch-xpu-ops/blob/main/.github/workflows/nightly_ondemand.yml#L25 
https://github.com/intel/torch-xpu-ops/blob/main/.github/workflows/nightly_ondemand_rolling.yml#L25 
https://github.com/intel/torch-xpu-ops/blob/main/.github/workflows/nightly_ondemand_whl.yml#L20

File: .github/workflows/_linux_ut.yml (hunk 2 above)
Comment (fengyuan14) [reply .github/workflows/_linux_ut.yml:98]: Yes, Just renaming them.

File: test/regressions/test_operation_on_device_1.py (hunk 5 above)
Comment (fengyuan14) [reply test/regressions/test_operation_on_device_1.py:6]: Yes.

**Review by chuanqi129 — APPROVED** [review]
LGTM. CC: @RUIJIEZHONG66166 @mengfei25 to aware this UT structure change

**Comment by fengyuan14** [comment]
> @fengyuan14 , by the way, please give some detailed information here.

Done

**Pull request merged**